web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -69,6 +175,40 @@ export class JobListItemComponent implements OnInit, OnDestroy {
         this.surveyId = id;
       })
     );
+
+    this.subscription.add(
+      this.loiService.getLocationsOfInterest$().pipe(shareReplay()).subscribe(lois => {
+        const jobLois = lois.filter((loi) => {
+          return loi.jobId === this.job?.id;
+        });
+
+        this.surveyService.getActiveSurvey$().subscribe((survey) => {
+          let numOfNoLabelLois = 0;
+          jobLois.forEach((loi) => {
+            this.submissionService.submissions$(survey, loi).subscribe((submissions) => {
```

Comment:
This created new subscriptions without disposing of previous ones. Instead, we you can `switchMap` to create a stream that disposes of the previous subscription each time a new item is emitted.